src/storage/firebase_upload_task_factory.ts

```diff
@@ -0,0 +1,26 @@
+import { FirebaseUploadTaskObservable } from './firebase_upload_task_observable';
+import { Observer } from 'rxjs/Observer';
+import { observeOn } from 'rxjs/operator/observeOn';
+import * as firebase from 'firebase/app';
+import 'firebase/storage';
+import * as utils from '../utils';
+import { UploadTask, UploadTaskSnapshot } from '../interfaces';
+
+export function FirebaseUploadTaskFactory (uploadTask: UploadTask): FirebaseUploadTaskObservable<UploadTaskSnapshot> {
+
+  const objectObservable = new FirebaseUploadTaskObservable((obs: Observer<UploadTaskSnapshot>) => {
```

Comment:
Yeah, I'll get this back up to speed with master to fix shortly. Was letting David get in his systemjs fixes before piling on.